### PR TITLE
Tooltips: Show BP modifiers from STAB and field effects

### PR DESCRIPTION
**Pull request by MacChaeger**
This is the last of the BP modifiers until I think of something else that I inevitably forgot.

### CODE REVIEW

File: js/client-battle-tooltips.js [hunk 1]
```diff
@@ -1205,12 +1229,56 @@ var BattleTooltips = (function () {
...
+		} else if (ability === 'levitate') {
+			isGrounded = false;
+		} else if (pokemonData.volatiles && (pokemonData.volatiles['magnetrise'] || pokemonData.volatiles['telekinesis'])) {
+			isGrounded = false;
+		} else if (!noItem && pokemonData.item !== 'airballoon') {
+			isGrounded = false;
+		} else if (!(pokemonData.volatiles && pokemonData.volatiles['roost'])) {
+			// If a Fire/Flying type uses Burn Up and Roost, it becomes ???/Flying-type, but it's still grounded.
+			for (var i = 0; i < types.length; i++) {
+				if (types[i] === 'Flying') isGrounded = false;
```
Comment (Zarel) [inline js/client-battle-tooltips.js]: > still grounded

> `isGrounded = false;`

Comment (MacChaeger) [reply js/client-battle-tooltips.js]: That just explains why it checks for roost before it checks the typing; I literally copied this straight out of `isGrounded` in the server's sim code, complete with the comment.

Comment (MacChaeger) [reply js/client-battle-tooltips.js]: That said, I don't actually think it's been tested what happens if a flying/??? type that has roosted also has levitate or something else that lifts it up...

**Comment by Zarel** [comment]
Looks good to me.

**Comment by MacChaeger** [comment]
OK well somebody remind me to change this when Marty confirms that that particular mechanic is [actually wrong](http://www.smogon.com/forums/posts/7503256/)

**Pull request merged**